dao/src/main/java/greencity/repository/UserRepo.java

```diff
@@ -120,25 +120,35 @@ public interface UserRepo extends JpaRepository<User, Long>, JpaSpecificationExe
      * @return List of friends.
      */
     @Query(nativeQuery = true, value = "SELECT * FROM ((SELECT user_id FROM users_friends AS uf "
-        + "WHERE uf.friend_id = :userId AND uf.status = 'FRIEND' AND "
-        + "(SELECT count(*) FROM habit_assign ha WHERE ha.habit_id = :habitId AND ha.user_id = uf.user_id "
-        + "AND ha.status = 'INPROGRESS') = 1) "
-        + "UNION "
-        + "(SELECT friend_id FROM users_friends AS uf "
-        + "WHERE uf.user_id = :userId AND uf.status = 'FRIEND' AND "
-        + "(SELECT count(*) FROM habit_assign ha WHERE ha.habit_id = :habitId AND ha.user_id = uf.friend_id "
-        + "AND ha.status = 'INPROGRESS') = 1)) as ui JOIN users as u ON user_id = u.id")
+            + "WHERE uf.friend_id = :userId AND uf.status = 'FRIEND' AND "
+            + "(SELECT count(*) FROM habit_assign ha WHERE ha.habit_id = :habitId AND ha.user_id = uf.user_id "
+            + "AND ha.status = 'INPROGRESS') = 1) "
+            + "UNION "
+            + "(SELECT friend_id FROM users_friends AS uf "
+            + "WHERE uf.user_id = :userId AND uf.status = 'FRIEND' AND "
+            + "(SELECT count(*) FROM habit_assign ha WHERE ha.habit_id = :habitId AND ha.user_id = uf.friend_id "
+            + "AND ha.status = 'INPROGRESS') = 1)) as ui JOIN users as u ON user_id = u.id")
     List<User> getFriendsAssignedToHabit(Long userId, Long habitId);
 
     /**
      * Get all user friends.
      *
      * @param userId The ID of the user.
-     *
      * @return list of {@link User}.
      */
     @Query(nativeQuery = true, value = "SELECT * FROM users WHERE id IN ( "
-        + "(SELECT user_id FROM users_friends WHERE friend_id = :userId and status = 'FRIEND')"
-        + "UNION (SELECT friend_id FROM users_friends WHERE user_id = :userId and status = 'FRIEND'));")
+            + "(SELECT user_id FROM users_friends WHERE friend_id = :userId and status = 'FRIEND')"
+            + "UNION (SELECT friend_id FROM users_friends WHERE user_id = :userId and status = 'FRIEND'));")
     List<User> getAllUserFriends(Long userId);
+
+    @Query(nativeQuery = true, value = "SELECT EXISTS (SELECT 1 FROM users_friends " +
+            "WHERE ((user_id = :userId AND friend_id = :friendId) OR (user_id = :friendId AND friend_id = :userId)) " +
```

Comment:
We save both userId -> friendId and friendId ->  userId connections for FRIENDS status. So we have no need to check combinations

Reply:
Fixed and tested out.
Thanks for notice